src/current.rs

```diff
@@ -35,12 +37,12 @@ pub fn clock_for(task_id: TaskId) -> VectorClock {
 
 /// Sets the `tag` field of the current task.
 /// Returns the `tag` which was there previously.
-pub fn set_tag_for_current_task(tag: Tag) -> Tag {
+pub fn set_tag_for_current_task(tag: Arc<dyn Debug>) -> Option<Arc<dyn Debug>> {
```

Comment:
Yeah, I'll swap to a trait. I think we want it to be `Hash` as well, so that we can store how many times something of tag X was scheduled. `Hash` is not object safe, but there are shenanigans one can do to work around. I chose `Arc` over `Box` as we don't need ownership of the tag, figured it would be cheaper to just pass references around. Most likely the user code will use the tag some number of times, and multiple tasks will have the same tag as well.

Reply:
Can we just use `Tag` directly and then implement `Tag` for `Arc<dyn Debug + ...>`?

Reply:
As in storing something ala: `counts: HashMap<Tag, usize>`? That's the idea, but there is no way to implement `Tag` for something which implements `Hash` as that is not object safe, ie, the following doesn't work:
```rust
pub trait Tag {}
trait NewTrait: Debug + std::hash::Hash + Eq + PartialEq {}
impl Tag for Arc<dyn NewTrait> {}
```
The only way I know which works (while in the `dyn Debug` world) is something ala:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b1f7cc3831363134eef1679cf1e8a75f

Reply:
Nevermind on the `Hash` stuff, we should be able to get what we need by storing on `TaskId`. Traited `Tag`.